sys/mips/malta/gt_pci.c

```diff
@@ -312,9 +312,9 @@ gt_pci_attach(device_t dev)
 	if (bus_space_map(sc->sc_st, IO_ICU2, 2, 0, &sc->sc_ioh_icu2) != 0)
 		device_printf(dev, "unable to map ICU2 registers\n");
 #else
-	sc->sc_ioh_elcr = MIPS_PHYS_TO_KSEG1(sc->sc_io + 0x4d0);
-	sc->sc_ioh_icu1 = MIPS_PHYS_TO_KSEG1(sc->sc_io + IO_ICU1);
-	sc->sc_ioh_icu2 = MIPS_PHYS_TO_KSEG1(sc->sc_io + IO_ICU2);
+	sc->sc_ioh_elcr = (vm_offset_t)MIPS_PHYS_TO_KSEG1(sc->sc_io + 0x4d0);
+	sc->sc_ioh_icu1 = (vm_offset_t)MIPS_PHYS_TO_KSEG1(sc->sc_io + IO_ICU1);
+	sc->sc_ioh_icu2 = (vm_offset_t)MIPS_PHYS_TO_KSEG1(sc->sc_io + IO_ICU2);
```

Comment:
Is this not implicit?

Reply:
Does a `vm_offset_t` cast DTRT against `caddr_t`?  I guess it doesn't need `__cheri_addr` in purecap?